sdks/python/apache_beam/io/requestresponse.py

```diff
@@ -411,3 +384,430 @@ def process(self, request: RequestT, *args, **kwargs):
   def teardown(self):
     self._metrics_collector.teardown_counter.inc(1)
     self._caller.__exit__(*sys.exc_info())
+
+
+class Cache(abc.ABC):
+  """Base Cache class for
+  :class:`apache_beam.io.requestresponse.RequestResponseIO`.
+
+  For adding cache support to RequestResponseIO, implement this class.
+  """
+  @abc.abstractmethod
+  def get_read(self):
+    """get_read returns a PTransform that reads from the cache."""
+    pass
+
+  @abc.abstractmethod
+  def get_write(self):
+    """get_write returns a PTransform that writes to the cache."""
+    pass
+
+  @abc.abstractmethod
+  def has_request_coder(self) -> bool:
```

Comment:
We can't use the cache. For enrichment handlers this will always be true since we use the string coder to encode. But for RRIO specific use cases, you need to provide both request and response coders for it to be usable.

Reply:
do we fail if users specify the cache but don't provide the coder?

Reply:
yes